paddle/fluid/pir/transforms/pd_op_to_kernel_pass.cc

```diff
@@ -776,416 +623,593 @@ phi::KernelKey GetKernelKey(
-    VLOG(6) << "kernel type " << kernel_key;
+  if (op_item->name() == "builtin.split") {
+    if (op_item->num_operands() > 0) {
+      for (size_t i = 0; i < op_item->num_operands(); ++i) {
+        auto cur_in = op_item->operand_source(i);
+        if (!cur_in) {
+          vec_inputs.emplace_back();
+          continue;
+        }
+        auto new_in = GetNewInput(cur_in, *map_value_pair, i, op_item->name());
+        vec_inputs.push_back(new_in);
 
-    if (op_item->name() == "pd_op.load_combine") {
-      kernel_key.set_dtype(phi::DataType::FLOAT32);
-    }
-    if (NeedFallBackCpu((op_item), kernel_fn_str, kernel_key)) {
-      kernel_key.set_backend(phi::Backend::CPU);
+        if (new_in.type().isa<pir::VectorType>()) {
+          auto vec_types = new_in.type().dyn_cast<pir::VectorType>().data();
+          for (uint64_t idx = 0; idx < vec_types.size(); idx++) {
+            op_output_types.push_back(vec_types[idx]);
+          }
+        } else {
+          PADDLE_THROW(
+              phi::errors::Unimplemented("only support vector type for now"));
+        }
+      }
     }
+  }
 
-    if (NeedFallBackFromGPUDNN2GPU(op_item, kernel_key)) {
-      kernel_key.set_backend(phi::Backend::GPU);
+  pir::OpInfo op_info = ctx->GetRegisteredOpInfo(op_item->name());
+  // Generate new op
+  pir::Operation* op = pir::Operation::Create(
+      vec_inputs, op_item->attributes(), op_output_types, op_info);
+  program->block()->push_back(op);
+  (*map_op_pair)[op_item] = op;
+  // only deal with single output
+  if (op_item->num_results() > 0) {
+    for (size_t i = 0; i < op_item->num_results(); ++i) {
+      (*map_value_pair)[op_item->result(i)] = op->result(i);
     }
+  }
+  VLOG(6) << "Deep copy a new builtin op: " << op_item->name();
+}
 
-    // only for single output
-    // need update new kernel key layout and data tyep
-
-    std::vector<pir::Type> op_output_types;
-    if (op_item->num_results() > 0) {
-      auto phi_kernel = phi::KernelFactory::Instance().SelectKernelWithGPUDNN(
-          kernel_fn_str, kernel_key);
-      auto args_def = phi_kernel.args_def();
-      auto output_defs = args_def.output_defs();
-      if (!UnchangeOutputOps.count(op_item->name()) &&
-          !IsLegacyOp(op_item->name())) {
-        PADDLE_ENFORCE_EQ(
-            op_item->num_results(),
-            output_defs.size(),
-            phi::errors::PreconditionNotMet(
-                "op [%s] kernel output args defs should equal op outputs",
-                op_item->name()));
-      }
+std::vector<pir::Type> BuildOpOutputType(pir::Operation* op_item,
+                                         const std::string& kernel_fn_str,
+                                         const phi::KernelKey& kernel_key,
+                                         pir::IrContext* ctx) {
+  if (op_item->num_results() == 0) {
+    return {};
+  }
+  std::vector<pir::Type> op_output_types;
+  auto phi_kernel = phi::KernelFactory::Instance().SelectKernelWithGPUDNN(
```

Comment:
```suggestion
  auto& phi_kernel = phi::KernelFactory::Instance().SelectKernelWithGPUDNN(
```
auto 与 auto& 是不同的，引用是不会自动推导的，可能会带来潜在copy